### ISSUE TITLE
Overtyping selected text and inserting snippet doesn't keep selected text as argument

### ISSUE THREAD

**Issue by kjshen** [James-Yu/LaTeX-Workshop#3531]
### Disable all the other extensions except for LaTeX Workshop, restart VS Code, and check that you still see this issue. [Required]

You still see this issue?: Yes

### Make sure to visit the [wiki FAQ](https://github.com/James-Yu/LaTeX-Workshop/wiki/FAQ) before filling an issue.

You visited the wiki?: Yes

## Describe the bug [Required]

You used to be able to put a selected formula into the equation environment with the LaTeX workshop's ```BEQ``` snippet by selecting the formula, typing BEQ and pressing tab/enter to select the snippet.
Similarly, snippets that complete a command and add curly brackets such as ```\section{}``` also used to be able to put the selected text into the curly brackets by start typing ```\sectio``` and then pressing tab/enter to select the snippet.
Currently, the selection is lost after inserting the snippet.
![env-snippet](https://user-images.githubusercontent.com/45091615/200649020-bd9b25f2-2da2-49ec-b381-89b515e8cb9c.gif)
![section-snippet](https://user-images.githubusercontent.com/45091615/200649039-b86264c0-a643-4da3-8100-fef224b618bb.gif)

This also used for work for custom snippets that had ```$TM_SELECTED_TEXT``` inside of them, but my custom snippets also don't work when overtyping my selection.

### To Reproduce

1. Select text
2. Type snippet name
3. Select snippet

### Expected behavior

The desired behavior corresponds to the "Surround Selection with LaTeX command" where the selected text is inserted inside the environment or the curly brackets:
![surround-with-env-snippet](https://user-images.githubusercontent.com/45091615/200649059-d7aed999-0802-4121-bc5c-e70c0a34f57b.gif)
![surround-with-section-snippet](https://user-images.githubusercontent.com/45091615/200649074-f4eb6750-bf67-4bda-98c6-abd4131b3112.gif)

### LaTeX Workshop Output [Required]

```
    [19:31:30] Initializing LaTeX Workshop.
    [19:31:30] Extension root: c:\Users\Kai\.vscode\extensions\james-yu.latex-workshop-9.0.1
    [19:31:30] $PATH: c:\program files\imagemagick-7.0.8-q16;c:\windows\system32;c:\windows;c:\windows\system32\wbem;c:\windows\system32\windowspowershell\v1.0\;c:\windows\system32\openssh\;c:\programdata\chocolatey\bin;c:\strawberry\c\bin;c:\strawberry\perl\site\bin;c:\strawberry\perl\bin;c:\program files\matlab\r2018b\bin;C:\Program Files (x86)\National Instruments\Shared\LabVIEW CLI;C:\Program Files (x86)\IVI Foundation\VISA\WinNT\Bin\;C:\Program Files\IVI Foundation\VISA\Win64\Bin\;C:\Program Files (x86)\IVI Foundation\VISA\WinNT\Bin;C:\Program Files\Git\cmd;C:\Android;C:\Windows\System32;C:\Program Files (x86)\Calibre2\;C:\Program Files\Microsoft SQL Server\120\Tools\Binn\;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files (x86)\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files\Intel\Intel(R) Management Engine Components\DAL;C:\Program Files\dotnet\;C:\Users\Kai\AppData\Local\Programs\Python\Python37-32\Scripts\;C:\Users\Kai\AppData\Local\Programs\Python\Python37-32\;C:\Users\Kai\AppData\Local\Microsoft\WindowsApps;C:\Strawberry\perl\bin;C:\Program Files\Inkscape;C:\Users\Kai\AppData\Local\Programs\Microsoft VS Code\bin;C:\Python27\;C:\Users\Kai\AppData\Local\Programs\MiKTeX 2.9\miktex\bin\x64\;
    [19:31:30] $SHELL: undefined
    [19:31:30] $LANG: undefined
    [19:31:30] $LC_ALL: undefined
    [19:31:30] process.platform: win32
    [19:31:30] process.arch: x64
    [19:31:30] vscode.env.appName: Visual Studio Code
    [19:31:30] vscode.env.remoteName: undefined
    [19:31:30] vscode.env.uiKind: 1
    [19:31:30] Configuration for workspace: undefined
    [19:31:30] editor.acceptSuggestionOnEnter: "on"
    [19:31:30] latex-workshop.bind.enter.key: true
    [19:31:30] latex-workshop.docker.enabled: false
    [19:31:30] latex-workshop.docker.image.latex: ""
    [19:31:30] latex-workshop.hover.preview.mathjax.extensions: []
    [19:31:30] latex-workshop.intellisense.package.enabled: true
    [19:31:30] latex-workshop.intellisense.update.aggressive.enabled: false
    [19:31:30] latex-workshop.intellisense.update.delay: 1000
    [19:31:30] latex-workshop.latex.autoBuild.run: "onFileChange"
    [19:31:30] latex-workshop.latex.build.forceRecipeUsage: true
    [19:31:30] latex-workshop.latex.outDir: "%DIR%"
    [19:31:30] latex-workshop.latex.recipes: [
    {
    "name": "latexmk",
    "tools": [
    "latexmk"
    ]
    },
    {
    "name": "pdflatex",
    "tools": [
    "pdflatex"
    ]
    },
    {
    "name": "biber",
    "tools": [
    "biber"
    ]
    },
    {
    "name": "pdflatex -> bibtex -> pdflatex*2",
    "tools": [
    "pdflatex",
    "bibtex",
    "pdflatex",
    "pdflatex"
    ]
    },
    {
    "name": "lualatex",
    "tools": [
    "lualatex"
    ]
    },
    {
    "name": "xelatex",
    "tools": [
    "xelatex"
    ]
    },
    {
    "name": "makeglossaries",
    "tools": [
    "makeglossaries"
    ]
    }
    ]
    [19:31:30] latex-workshop.latex.tools: [
    {
    "name": "latexmk",
    "command": "latexmk",
    "args": [
    "-shell-escape",
    "-synctex=1",
    "-interaction=nonstopmode",
    "-file-line-error",
    "-pdf",
    "%DOC%"
    ]
    },
    {
    "name": "pdflatex",
    "command": "pdflatex",
    "args": [
    "-shell-escape",
    "-synctex=1",
    "-interaction=nonstopmode",
    "-file-line-error",
    "%DOC%"
    ]
    },
    {
    "name": "lualatex",
    "command": "lualatex",
    "args": [
    "-synctex=1",
    "-interaction=nonstopmode",
    "--shell-escape",
    "-file-line-error",
    "%DOC%"
    ]
    },
    {
    "name": "bibtex",
    "command": "bibtex",
    "args": [
    "%DOCFILE%"
    ]
    },
    {
    "name": "biber",
    "command": "biber",
    "args": [
    "%DOCFILE%"
    ]
    },
    {
    "name": "xelatex",
    "command": "xelatex",
    "args": [
    "%DOCFILE%"
    ]
    },
    {
    "name": "makeglossaries",
    "command": "makeglossaries",
    "args": [
    "%DOCFILE%"
    ]
    }
    ]
    [19:31:30] latex-workshop.viewer.pdf.internal.keyboardEvent: "auto"
    [19:31:30] Creating a new file watcher.
    [19:31:30] watcherOptions: {"useFsEvents":false,"usePolling":false,"interval":300,"binaryInterval":1000,"awaitWriteFinish":{"stabilityThreshold":250}}
    [19:31:30] Creating PDF file watcher.
    [19:31:30] watcherOptions: {"useFsEvents":false,"usePolling":false,"interval":300,"binaryInterval":1000,"awaitWriteFinish":{}}
    [19:31:30] Creating Bib file watcher.
    [19:31:30] watcherOptions: {"useFsEvents":false,"usePolling":false,"interval":300,"binaryInterval":1000,"awaitWriteFinish":{"stabilityThreshold":250}}
    [19:31:30] Set $LATEXWORKSHOP_DOCKER_LATEX: ""
    [19:31:30] pdflatex is provided by MiKTeX
    [19:31:30] [Server] Creating LaTeX Workshop http and websocket server.
    [19:31:30] LaTeX Workshop initialized.
    [19:31:30] Trigger characters for intellisense of LaTeX documents: ["\\",",","{"]
    [19:31:30] Bibtex format config: {"tab":"  ","case":"lowercase","left":"{","right":"}","trailingComma":false,"sort":["key"],"alignOnEqual":true,"sortFields":false,"fieldsOrder":[],"firstEntries":["string","xdata"]}
    [19:31:30] Current workspace folders: undefined
    [19:31:30] Found root file from active editor: c:\Users\Kai\sciebo\LaTeX\MWE.tex
    [19:31:30] Root file changed: from undefined to c:\Users\Kai\sciebo\LaTeX\MWE.tex
    [19:31:30] Start to find all dependencies.
    [19:31:30] Root file languageId: latex
    [19:31:30] Reset file watcher.
    [19:31:30] Parsing a file and its subfiles: c:\Users\Kai\sciebo\LaTeX\MWE.tex
    [19:31:30] Parse fls file.
    [19:31:30] Fls file found: c:\Users\Kai\sciebo\LaTeX\MWE.fls
    [19:31:30] Parse aux file: c:\Users\Kai\sciebo\LaTeX\MWE.aux
    [19:31:30] [Server] Server successfully started: {"address":"127.0.0.1","family":"IPv4","port":56909}
    [19:31:30] Added to file watcher: c:\Users\Kai\sciebo\LaTeX\MWE.tex
    [19:31:30] Snippet data loaded.
    [19:31:30] [Server] valdOrigin is http://127.0.0.1:56909
    [19:31:30] Checking for duplicate labels: c:\Users\Kai\sciebo\LaTeX\MWE.tex.
    [19:31:32] Manager.fileWatcher.getWatched: {"c:\\Users\\Kai\\sciebo\\LaTeX":["MWE.tex"]}
    [19:31:32] Manager.filesWatched: ["c:\\Users\\Kai\\sciebo\\LaTeX\\MWE.tex"]
    [19:31:32] BibWatcher.bibWatcher.getWatched: {}
    [19:31:32] BibWatcher.bibsWatched: []
    [19:31:32] PdfWatcher.pdfWatcher.getWatched: {}
    [19:31:32] PdfWatcher.pdfsWatched: []
    [19:31:32] PdfWatcher.watchedPdfVirtualUris: []
    [19:31:32] PdfWatcher.ignoredPdfUris: []
    [19:34:15] File watcher - file changed: c:\Users\Kai\sciebo\LaTeX\MWE.tex
    [19:34:15] Parsing a file and its subfiles: c:\Users\Kai\sciebo\LaTeX\MWE.tex
    [19:34:15] Checking for duplicate labels: c:\Users\Kai\sciebo\LaTeX\MWE.tex.
    [19:34:15] Auto build started detecting the change of a file: c:\Users\Kai\sciebo\LaTeX\MWE.tex
    [19:34:15] BUILD command invoked.
    [19:34:15] The document of the active editor: file:///c:/Users/Kai/sciebo/LaTeX/MWE.tex
    [19:34:15] The languageId of the document: latex
    [19:34:15] Building root file: c:\Users\Kai\sciebo\LaTeX\MWE.tex
    [19:34:15] Build root file c:\Users\Kai\sciebo\LaTeX\MWE.tex
    [19:34:15] outDir: c:/Users/Kai/sciebo/LaTeX
    [19:34:15] Recipe step 1: latexmk
    [19:34:15] Recipe step 1 args: ["--max-print-line=10000","-shell-escape","-synctex=1","-interaction=nonstopmode","-file-line-error","-pdf","c:/Users/Kai/sciebo/LaTeX/MWE"]
    [19:34:15] Recipe step env: undefined
    [19:34:15] Recipe step root file: c:\Users\Kai\sciebo\LaTeX\MWE.tex
    [19:34:15] cwd: c:\Users\Kai\sciebo\LaTeX
    [19:34:15] LaTeX build process spawned. PID: 11308.
    [19:34:16] A step in recipe finished. PID: 11308.
    [19:34:16] Successfully built c:\Users\Kai\sciebo\LaTeX\MWE.tex.
    [19:47:49] LOG command invoked: default
```

## Desktop [Required]

 - OS: Windows 10
 - VS Code version: 1.73.0
 - Extension version: 9.0.1 (I noticed this behavior during 8.29.0 as well if I remember correctly)

**Comment by James-Yu** [comment]
This is unlikely an extension issue. I downgraded to 8.27.0 and still observed this issue. I recommend you to test snippets for other languages, and report to vscode if the problem persisted.

**Comment by kjshen** [comment]
I found the corresponding issue in the VSCode Project https://github.com/microsoft/vscode/issues/165746

**Comment by kjshen** [comment]
> I found the corresponding issue in the VSCode Project [microsoft/vscode#165746](https://github.com/microsoft/vscode/issues/165746)

It appears that with today's update (November 2022 (version 1.74)), this issue has been fixed partially, meaning snippets of the form

    "mycmd": {"prefix": "\\mycmd","body": "\\mycmd{$TM_SELECTED_TEXT$1}",},
    
will replace `TM_SELECTED_TEXT` with the text that was overtyped.
However, if the backslash in the prefix is left out, i.e.

    "mycmd": {"prefix": "mycmd","body": "\\mycmd{$TM_SELECTED_TEXT$1}",},

it does not work for me. Consequently, all environment snippets such as `BEQ` are still not working, which I assume are of the form

    "mybeq": {
        "prefix": "mybeq",
        "body": [
            "\\begin{equation}",
            "    $TM_SELECTED_TEXT$1",
            "\\end{equation}",
        ],
    },
    
 As a workaround, custom snippets of that form will work by adding the backslash to the prefix, i.e.

    "mybeq": {
        "prefix": "\\mybeq",
        "body": [
            "\\begin{equation}",
            "    $TM_SELECTED_TEXT$1",
            "\\end{equation}",
        ],
    },

**Comment by kjshen** [comment]
I found that my statement above is only true for .tex files.
After defining global snippets, I found the opposite to be true in .py files:
Snippets that have a ```\\``` in the prefix do not work with ```$TM_SELECTED_TEXT$1```, while snippets without  ```\\```  in the prefix work.

> This is unlikely an extension issue. I downgraded to 8.27.0 and still observed this issue. I recommend you to test snippets for other languages, and report to vscode if the problem persisted.

Would that mean that making environment snippets work with overtyping text again is now an extension issue? @James-Yu
